common/src/main/templates/io/netty/util/collection/KObjectHashMap.template

```diff
@@ -351,8 +353,7 @@ public class @K@ObjectHashMap<V> implements @K@ObjectMap<V>, Iterable<@K@ObjectM
      * Returns the hashed index for the given key.
      */
     private int hashIndex(@k@ key) {
-        // Allowing for negative keys by adding the length after the first mod operation.
-        return (hashCode(key) % keys.length + keys.length) % keys.length;
+        return hashCode(key) & mask;
```

Comment:
@buchgr - Not sure where you are with the headers optimization stuff...but if [DefaultHeaders.index](https://github.com/netty/netty/blob/master/codec/src/main/java/io/netty/handler/codec/DefaultHeaders.java#L1228) still needs to exist we should investigate making a similar change there.